src/x86/nt.rs

```diff
@@ -0,0 +1,275 @@
+//! Non-temporal store operations.
+use core::{marker::PhantomData, ptr};
+
+#[cfg(target_arch = "x86")]
+use core::arch::x86::{
+    self as arch, __m128, __m128d, __m128i, __m256, __m256d, __m256i, _mm_sfence,
+};
+#[cfg(target_arch = "x86_64")]
+use core::arch::x86_64::{
+    self as arch, __m128, __m128d, __m128i, __m256, __m256d, __m256i, _mm_sfence,
+};
+
+/// Load from a 32-bit aligned address with non-temporal hint, avoiding filling the cache.
+#[inline]
+#[cfg(any())]
+#[target_feature(enable = "avx2")]
+pub fn _mm256_stream_load_si256(addr: &__m256i) -> __m256i {
+    unsafe { arch::_mm256_stream_load_si256(addr) }
+}
+
+/// Store into a 32-bit aligned address with non-temporal hint, avoiding clobbering the cache.
+#[inline]
+#[target_feature(enable = "avx")]
+pub fn _mm256_stream_store_256i(addr: &mut NonTemporalStoreable<'_, __m256i>, v: __m256i) {
+    unsafe { arch::_mm256_stream_si256(addr.inner.as_ptr(), v) }
+}
+
+/// Store a 128-bit floating point vector of `[2 × double]` into a 128-bit aligned memory location.
+/// To minimize caching, the data is flagged as non-temporal (unlikely to be used again soon).
+#[inline]
+#[target_feature(enable = "sse2")]
+pub fn _mm_stream_pd(addr: &mut NonTemporalStoreable<'_, __m128d>, v: __m128d) {
+    unsafe { arch::_mm_stream_pd(addr.inner.as_ptr() as *mut f64, v) }
```

Comment:
```suggestion
    unsafe { arch::_mm_stream_pd(addr.inner.as_ptr().cast(), v) }
```
For these and other `as` pointer casts if possible.